src/Http/Routing/test/UnitTests/Matching/DfaMatcherConformanceTest.cs

```diff
@@ -23,7 +24,132 @@ public override async Task Quirks_CatchAllParameter(string template, string path
             MatcherAssert.AssertMatch(httpContext, endpoint, keys, values);
         }
 
+        // https://github.com/dotnet/aspnetcore/issues/18677
+        [Theory]
+        [InlineData("/middleware", 1)]
+        [InlineData("/middleware/test", 1)]
+        [InlineData("/middleware/test1/test2", 1)]
+        [InlineData("/bill/boga", 0)]
+        public virtual async Task Match_Regression_1867_CorrectBehavior(string path, int endpointIndex)
```

Comment:
nit: These tests don't need to be virtual. I don't think we'd ever inherit from DfaMatcherConformanceTest